### PR TITLE
Added documentation for aggregated parameters

### LINKED ISSUE #273: Documentation: aggregated parameters
AggregatedParameter: sum, product, max, min, mean, custom
AggregatedIndexParameter: sum, max, min, any, all, custom
## Example usage
- [sum] Adding supplies or demands together (i.e. one node instead of two)
- [product] Complex demands (see demand saving)
- [max] Worst-case demand saving level between two (or more) reservoirs
- [any / all] As above, with different condition.

Custom function only available via Python (not JSON).

Also worth mentioning MaxParameter/MinParameter/NegativeParameter as these are optimised replacements for common AggregatedParameter tasks.

### PR DESCRIPTION
**Pull request by snorfalorpagus**
This assumes #414 will be merged.

Closes #273.

### CODE REVIEW

**Comment by snorfalorpagus** [comment]
Need to add `NegativeMaxParameter` to this.

**Comment by snorfalorpagus** [comment]
I think this is ready to be merged.

File: docs/source/cookbook/aggregated_parameter.rst [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+Sum, Min, Max
+=============
+
+The ``"sum"``, ``"min"`` and ``"max"`` aggregation functions are available to both aggregated value and aggregated index parameters.
+
+Product, Mean, Median
+=====================
+
+The ``"product"``, ``"mean"`` and ``"median"`` aggregation functions are only available for aggregated value parameters, as they could return non-integer values.
```
Comment (jetuk) [inline docs/source/cookbook/aggregated_parameter.rst]: This isn't true for product?

File: docs/source/cookbook/aggregated_parameter.rst [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+An example use is an index parameter which is "on" when any reservoir in a group is below its control curve.
+
+Custom aggregation functions
+============================
+
+Custom aggregation functions can be used via the Python API only. The function is called with the values of the individual parameters and should return the aggregated value. For example, the following aggregation function returns the 25th percentile of the values:
+
+.. code-block:: python
+
+    func = lambda *x: np.percentile(x, 25)
```
Comment (jetuk) [inline docs/source/cookbook/aggregated_parameter.rst]: is the * meant to be there?

Comment (snorfalorpagus) [inline docs/source/cookbook/aggregated_parameter.rst]: I think I put it there intentionally, but I don't think it's needed.

**Comment by snorfalorpagus** [comment]
Addressed comments. I've added the product function to `AggregatedIndexParameter`.

**Pull request merged**